From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lew <skycubesv@gmail.com>
Date: Fri, 10 Jun 2022 18:00:45 -0300
Subject: [PATCH] Cache chunk coordinations

Prevent the creation of multiple Chunk Coordinations by calling the Chunk#j method

diff --git a/src/main/java/net/minecraft/server/Chunk.java b/src/main/java/net/minecraft/server/Chunk.java
index bfeababe859d5ed658bf8afe5796cf49a4e38626..592025a50da46dbe7a4b51bdde097560332a2e39 100644
--- a/src/main/java/net/minecraft/server/Chunk.java
+++ b/src/main/java/net/minecraft/server/Chunk.java
@@ -130,6 +130,8 @@ public class Chunk {
     }
     // CraftBukkit end
 
+    private final ChunkCoordIntPair chunkCoords; // PandaSpigot
+
     public Chunk(World world, int i, int j) {
         this.sections = new ChunkSection[16];
         this.e = new byte[256];
@@ -155,6 +157,7 @@ public class Chunk {
         if (!(this instanceof EmptyChunk)) {
             this.bukkitChunk = new org.bukkit.craftbukkit.CraftChunk(this);
         }
+        this.chunkCoords = new ChunkCoordIntPair(this.locX, this.locZ); // PandaSpigot
     }
 
     public org.bukkit.Chunk bukkitChunk;
@@ -1204,7 +1207,7 @@ public class Chunk {
     }
 
     public ChunkCoordIntPair j() {
-        return new ChunkCoordIntPair(this.locX, this.locZ);
+        return this.chunkCoords; // PandaSpigot
     }
 
     public boolean c(int i, int j) {
